app/services/user_signup.rb

```diff
@@ -0,0 +1,49 @@
+class UserSignup
+
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        :amount => 999,
+        :source => stripe_token, # obtained with Stripe.js
+        :description => "Sign up charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation(invitation_token)
+        MyflixMailer.notify_on_signup(@user).deliver
+        @status = :success
+        self
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "Invalid user information. Please check the errors below."
+      self
+    end
+  end
+
+  def successful?
+    @status == :success
+  end
+ 
+  private
+
+  def handle_invitation(invitation_token)
+    if invitation_token.present?
+        invitation = Invitation.where(token: invitation_token).first
```

Comment:
Instead of `where`...`first`, you can use `find_by` with the same hash argument.